dj_database_url/__init__.py

```diff
@@ -172,10 +172,13 @@ def parse(
             options["ssl"] = {"ca": values[-1]}
             continue
 
-        try:
-            options[key] = int(values[-1])
-        except (TypeError, ValueError):
-            options[key] = values[-1]
+        value = values[-1]
+        if value.isdigit():
+            options[key] = int(value)
+        elif value in ("true", "false"):
```

Comment:
Possibly `value.lower()` to support upper-case/title-case variants? And the same for the comparison?

Reply:
I noticed that Python 3.7 is nearing its end of life in June 2023.  Given this, would it be feasible for us to consider upgrading to Python 3.8 or later? It will enable us to leverage Assignment Expressions (PEP 572).

```python
elif (value_lowercase := value.lower()) in ("true", "false"):
    options[key] = value_lowercase == "true"
````